enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,910 @@
+8. Configuration for setting or deleting a header on a Route will override configuration for a header with the same name on the IngressController.
+
+
+### Non-Goals
+
+1. Validation that each specified value is valid for the respective header (e.g. require that either "DENY" or "SAMEORIGIN" is 
+   specified for [the `x-frame-options` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options)).
+2. Setting load balancing algorithm using header values. The HAproxy configuration provides a [load balancing algorithm](http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-balance) which relies on the
+   check against the incoming request's header values. Even though this enhancement adds the header manipulation capability it
+   doesn't aim at providing (or enhancing) the load balancing API for the IngressController or Route.  Therefore such a 
+   load balancing algorithm is out of scope for this enhancement and should be handled with a separate RFE.
+3. Configuring buffer sizes for headers (e.g. customizing HAProxy's `tune.bufsize` and `tune.maxrewrite` settings).
+4. Modifying headers for a passthrough route. 
+5. Providing dedicated API knobs for the [appending](http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-response%20add-header) or 
+   [replacing](http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-response%20replace-header) headers. 
+   The headers will be modified according to the [set-header HAProxy configuration stanza](http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-http-response%20set-header): add header 
+   if it doesn't exist and remove before adding if it does.
+6. Allowing sample fetchers other than `req.hdr`, `res.hdr`, and `ssl_c_der` or converters other than `lower` and `base64` in dynamic header values.
+7. Providing a way to prohibit the Route owner from overriding the configuration for an IngressController if the Route and IngressController both set or delete a header with the same name.
+8. The http request headers X-SSL, X-SSL-Client-Verify, X-SSL-Client-Serial, X-SSL-Client-Version, X-SSL-Client-SHA1, X-SSL-Client-DN, X-SSL-Client-CN, X-SSL-Issuer, X-SSL-Client-NotBefore, X-SSL-Client-NotAfter, X-SSL-Client-DER which are
+   set by the haproxy template can be over-ridden or deleted as we don't blacklist them. This EP doesn't have the ability to disallow this. However, we don't recommend changing these values.
+9. We explicitly delete Proxy header in the haproxy template. This header can be added using this API as it is not put in the blacklist. So this EP doesn't have the ability to disallow this. However, we don't recommend adding this header using this API.
+10. The `HOST` header is made sure that it is in lowercase. Using this API one might be able change this to upper-case as this EP doesn't have the ability to disallow this. However, we don't recommend this.
```

Comment:
Thanks for addressing this.  I see this as needing more details than just listing as non-goals.  I think you need to address why you chose **not** to add these headers to the same "deny list" that you added the HSTS header to.  Add these to the same section in which you discuss why you added the HSTS header.  

Also, the company prefers the phrases "deny list" and "approved list" rather than blacklist and whitelist because they have the same meaning to everyone, and a better technical connotation.

Reply:
done